transactionparser/src/finalbalances.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+var _ = require('lodash');
+var normalizeNodes = require('./utils').normalizeNodes;
+var dropsToXRP = require('./utils').dropsToXRP;
+var groupByAddress = require('./utils').groupByAddress;
+var parseValue = require('./utils').parseValue;
+
+
+function parseFinalBalance(node) {
+  if (node.newFields.Balance) {
+    return parseValue(node.newFields.Balance);
+  } else if (node.finalFields.Balance) {
+    return parseValue(node.finalFields.Balance);
+  }
+  return null;
+}
+
+function parseXRPFinalBalance(node) {
+  var finalBalance = parseFinalBalance(node);
+
+  if (finalBalance === null) {
+    return null;
+  }
+
+  return {
+    address: node.finalFields.Account || node.newFields.Account,
+    balance: {
+      counterparty: '',
+      currency: 'XRP',
+      value: dropsToXRP(finalBalance).toString()
+    }
+  };
+}
+
+function parseTrustlineFinalBalance(node) {
+  var finalBalance = parseFinalBalance(node);
+
+  if (finalBalance === null) {
+    return null;
+  }
+
+  /*
+   * A trustline can be created with a non-zero starting balance
+   * If an offer is placed to acquire an asset with no existing trustline,
+   * the trustline can be created when the ofer is taken.
+   */
+  var fields = _.isEmpty(node.newFields) ? node.finalFields : node.newFields;
+
+  // the balance is always from low node's perspective
+  return {
+    address: fields.LowLimit.issuer,

```

Comment:
In metadata we don't have information about final balance of other end of trustline